src/team_lib/ucx/xccl_ucx_lib.c

```diff
@@ -108,12 +114,60 @@ static ucs_config_field_t xccl_tl_ucx_context_config_table[] = {
     {NULL}
 };
 
+static size_t xccl_ucx_get_src_buffer_len(xccl_coll_op_args_t *coll_args, int team_size)
```

Comment:
maybe have 1 implementation for both xccl_ucx_get_src/dst buffer_len to reduce code duplication? (the fn would have 1 more argument is_src ? )